sport/app/football/feed/Competitions.scala

```diff
@@ -101,6 +101,9 @@ trait Competitions extends implicits.Football {
 
   def matches: Seq[FootballMatch] = competitions.flatMap(_.matches)
 
+  def isAMatchInProgress(matches: Seq[FootballMatch], clock: Clock): Boolean =
+    matches.exists(game => game.isLive || game.date.minusMinutes(5).toLocalDate.isBefore(LocalDate.now(clock)))
```

Comment:
One thing to be aware of with the implementation as written, is that it will cause this job to re-run all day after the match has finished. We check if it is live, or if it `the start time + 5 minutes is before the current time`, which will be true all day after the match starts.

Changing this is going to be tricky! There are lots of edge cases around the start and end times of football matches, which is probably why we just use isLive at the moment. Sometimes matches are rescheduled or postponed. Sometimes they start late. Sometimes they have a lot of added injury time, or go to extra time and penalties. Sometimes the match is paused for a few hours in the middle and restarted later that day, or abandoned altogether halfway through. And so on!

The current implementation is not a silly one, so I'm not adding this comment to say it is wrong! Just checking it is what you mean. If we accept that we're going to be regularly refreshing the matchDay endpoint all day after any matches start, then that's probably fine.

Definitely something to be aware of though, and I'd suggest adding some documentation this function that describes the trade-offs that have been considered in this implementation. This documentation could be a comment, but could also be a good test suite that not only describes, but also checks and enforces, the desired behaviour.

Certainly either way, it'd be important to have a few tests for this fiddly logic, as we chatted about earlier :-)

Reply:
Having read this more carefully, I think this may be wrong!

It looks like at the moment, this will be passed *all* the matches:
https://github.com/guardian/frontend/pull/24605/files#diff-aa67d62dcc82d6d0c6c1b541ab0d34f0f6ef461f67e08698bb3e4a4e2c1e37e0R413

Crucially, this isn't *today's* matches. This is literally all the matches! So (for example) every match in the Premier league this season, including all the results for previous weeks and fixtures for the weeks ahead.

That means this will (almost) always return true, because there will (almost) always be a match that started before (now - 5 minutes).

Something like `withTodaysMatches` might be your answer, but you'll have to read the code carefully and apply your knowledge of football to think about what each of these datasets means. Don't forget that not all matches happen in the UK, so 'match day' might be a bit confusing in a country with a timezone that means the date (in London) changes midway through a running match.

Sport is very complicated! Whether or not this code could be improved (and it certainly could), as we see here a lot of the complexity we're dealing with comes from the fact that sport data is quite complex and on top of that there are a lot of edge cases to consider.

Here is another example of how using `isLive` and ignoring the date logic might make a lot of this complexity vanish for your use-case. It depends what you're after, and in particular whether you are prepared to drop the first 5 minutes of updates.

Reply:
Great explanation, thanks a lot. Reading this, made me question and challenge myself :) There are 4 points I can think of. 

1- My logic is wrong. 
`matches.exists(game => game.isLive || game.date.minusMinutes(5).toLocalDate.isBefore(LocalDate.now(clock))` 

What I wanted to achieve was to find out if there’s a game that’s about to start (5 minutes to start) so that we can make `refreshMatchDay` job more frequent 5 minutes before the game starts, but I didn't want it to continue for the whole day. 
Here in the wrong code it’s just checking if current time is after (game - 5 minutes) while it had to check if current time is after (game start time - 5 minutes) and before (game start time + 5 minutes).

If I’m not wrong, this way, it returns true 5 minutes before the game until 5 minutes after the game. A 10 minute buffer to make up the the time it can take until isLive returns true for a game. 

2- As you said I should write some tests to be able to make sure the date logic is working as expected. Will do it and let you know when it’s ready.

3- You mentioned about the list of matches that includes all fixtures, results and matchDays. Do you think we should not do the check for all the matches (including MatchDay, Fixtures, Results). Could it be better to just filter the matches and get the MatchDay list?


4- Something I can’t know for sure is that, we are calling PA `/matchDay/` endpoint with a local date. We calculate that local date based on the defaultClock which is the system default zone (I assume that’s the server time zone which is Irland Greenwhich?)
```
val defaultClock = Clock.systemDefaultZone()
LocalDate.now(defaultClock)
```
We are passing the local date of UK time (if server is Ireland), but how does PA understand this? Does PA consider it as uk date? Or does it consider as the date of the country in which the game is played? I couldn't find out from their documentation http://developer.press.net/io-docs

Reply:
@adamnfish What's your reason for suggesting `withTodaysMatches`? Is it because of the potential performance hit of iterating through *all* matches? or because, with the logic as it currently stands we would be assuming potentially every match is live? If its the second, we should be able to resolve this with Marjan's new logic.

`withTodaysMatches` certainly sounds interesting but taking a look at the code, it's a bit confusing. It looks like the logic would fail if the match date is the same date as today. There aren't any tests for this, so we're going to write some now to check that this logic is working correctly.

Reply:
Also, given that we currently only use `isLive`, does that mean we currently accept there is the change there will be no updates for the first 5 minutes?

Reply:
This issue was discussed in the call. As a result, the date logic was updated and we are passing `matches` ( the whole list of matches) to the `isMatchLiveOrAboutToStart`.